opendata_collector_module/metrics_opendata_collector/opendata_collector.py

```diff
@@ -20,6 +20,171 @@
 #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #  THE SOFTWARE.
 
+import datetime
+from operator import itemgetter
+from typing import Any, Dict, List, Tuple
 
-def collect_opendata(settings: dict, source_settings: dict):
-    pass
+import requests
+
+from metrics_opendata_collector.constants import DT_FORMAT_WO_TZ
+from metrics_opendata_collector.logger_manager import LoggerManager
+from metrics_opendata_collector.mongodb_manager import MongoDbManager
+from metrics_opendata_collector.opendata_api_client import (
+    InputValidationError, OpenDataAPIClient)
+from metrics_opendata_collector.settings import MetricsSettingsManager
+
+from . import __version__
+
+
+def collect_opendata(source_id: str, settings_manager: MetricsSettingsManager) -> None:
```

Comment:
Seems quite a large and heavy function... maybe it would make sense to have it little bit divided into smalled functions? Like one for handling rows. It could include functions from within this file, but a separation would really help with the readability of this function

Reply:
Thanks. Did some refactoring to improve it.